Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp

```diff
@@ -0,0 +1,800 @@
+
+    auto status() const { return cmp.status; }
+    auto weight() const { return cmp.weight; }
+    auto charge() const { return cmp.state.q; }
+    auto pathAccumulated() const { return cmp.state.pathAccumulated; }
+    const auto& pars() const { return cmp.state.pars; }
+    const auto& derivative() const { return cmp.state.derivative; }
+    const auto& jacTransport() const { return cmp.state.jacTransport; }
+    const auto& cov() const { return cmp.state.cov; }
+    const auto& jacobian() const { return cmp.state.jacobian; }
+    const auto& jacToGlobal() const { return cmp.state.jacToGlobal; }
+
+    template <typename propagator_state_t>
+    auto singleState(const propagator_state_t& state) const {
+      using DeducedStepping = decltype(state.stepping.components.front().state);
+      static_assert(std::is_same_v<SingleState, DeducedStepping>);
+
+      return SinglePropState<
+          const SingleState, const decltype(state.navigation),
+          const decltype(state.options), const decltype(state.geoContext)>(
+          cmp.state, state.navigation, state.options, state.geoContext);
+    }
+
+    const auto& singleStepper(const MultiEigenStepperLoop& stepper) const {
+      return static_cast<const SingleStepper&>(stepper);
+    }
+  };
+
+  /// A proxy struct which allows access to a single component of the
+  /// multi-component state. It has the semantics of a const reference, i.e.
+  /// it requires a const reference of the single-component state it
+  /// represents
+  using ConstComponentProxy =
+      ComponentProxyBase<const typename State::Component>;
+
+  /// A proxy struct which allows access to a single component of the
+  /// multi-component state. It has the semantics of a mutable reference, i.e.
+  /// it requires a mutable reference of the single-component state it
+  /// represents
+  struct ComponentProxy : ComponentProxyBase<typename State::Component> {
+    using Base = ComponentProxyBase<typename State::Component>;
+
+    using Base::charge;
+    using Base::cmp;
+    using Base::cov;
+    using Base::derivative;
+    using Base::jacobian;
+    using Base::jacToGlobal;
+    using Base::jacTransport;
+    using Base::pars;
+    using Base::pathAccumulated;
+    using Base::singleState;
+    using Base::singleStepper;
+    using Base::status;
+    using Base::weight;
+
+    const State& all_state;
+
+    ComponentProxy(typename State::Component& c, const State& s)
+        : Base(c), all_state(s) {}
+
+    auto& status() { return cmp.status; }
+    auto& weight() { return cmp.weight; }
+    auto& charge() { return cmp.state.q; }
+    auto& pathAccumulated() { return cmp.state.pathAccumulated; }
+    auto& pars() { return cmp.state.pars; }
+    auto& derivative() { return cmp.state.derivative; }
+    auto& jacTransport() { return cmp.state.jacTransport; }
+    auto& cov() { return cmp.state.cov; }
+    auto& jacobian() { return cmp.state.jacobian; }
+    auto& jacToGlobal() { return cmp.state.jacToGlobal; }
```

Comment:
Do we really need `auto` keywords here everywhere - I think these are explicit types, aren't they ?

Reply:
It's not against, it sort of hides whether the return type is e.g. a `Object&` or a `const Object&` which has a big consequence to the caller, I just wanted tot make this clear. That can be just done with a comment, then we're good, as they are accessors it's clear then.

Reply:
Yeah I see that it might be not quite clear, that the `const`-qualified accessors are inherited from the `ComponentProxyBase`. I did this to avoid code duplications between the const and the non-const `ComponentProxy`...

I'll add a comment to clarify this.